docs/src/test/java/docs/http/javadsl/HttpClientExampleDocTest.java

```diff
@@ -208,9 +209,26 @@ public void testSingleRequestWithHttpsProxyExampleWithAuth() {
                   HttpRequest.create("https://github.com"),
                   Http.get(system).defaultClientHttpsContext(),
                   poolSettingsWithHttpsProxy, // <- pass in the custom settings here
-                  system.log(),
-                  materializer);
+                  system.log());
 
     //#auth-https-proxy-example-single-request
   }
+
+  // compile only test
+  public void testCollectingHeadersExample() {
+
+    final ActorSystem system = ActorSystem.create();
+
+    //#collecting-headers-example
+    final CompletionStage<HttpResponse> responseFuture =
+      Http.get(system)
+        .singleRequest(HttpRequest.create("http://akka.io"));
+
+      responseFuture.thenApply(response -> StreamSupport.stream(response.getHeaders().spliterator(), false)
+        .filter(SetCookie.class::isInstance)
+        .map(SetCookie.class::cast)
+        .collect(Collectors.toList())
+      ).thenAccept(cookies -> System.out.println("Cookies set by a server: " + cookies));
```

Comment:
I guess it's worth to discard the entity bytes in this case. Just in case somebody copy-pastes this example and then wonders why the pool is full and can't issue any more requests